main.py

```diff
@@ -1,87 +1,136 @@
- 
-
-from influxdb import InfluxDBClient
-
+from statistics import mean
 from datetime import datetime
+from typing import List
 
-client = InfluxDBClient()
+import contextlib
+
+
+
+# client = InfluxDBClient()
 
-from statistics import mean
 
-current_wattages = []
-
-def add_wattage(watt_current: float, watt_total_low: float, watt_total_high: float) -> None:
-  global current_wattages
-  current_date = datetime.now().isoformat()
-  current_wattages.append(watt_current)
-  if len(current_wattages) == 10:
-    # done with a minute, lets write
-    avg = mean(current_wattages)
-    json_body = [
-      {
-          "measurement": "energy",
-          "time": current_date,
-          "fields": {
-              "watt_current": float(avg),
-              "watt_total_low": watt_total_low,
-              "watt_total_low": watt_total_high
-          }
-      }
-    ]
-    current_wattages = []
-    client.write_points(json_body)
+# current_wattages = []
 
+# def add_wattage(watt_current: float, watt_total_low: float, watt_total_high: float) -> None:
+#   global current_wattages
+#   current_date = datetime.now().isoformat()
+#   current_wattages.append(watt_current)
+#   if len(current_wattages) == 10:
+#     # done with a minute, lets write
+#     avg = mean(current_wattages)
+#     json_body = [
+#       {
+#           "measurement": "energy",
+#           "time": current_date,
+#           "fields": {
+#               "watt_current": float(avg),
+#               "watt_total_low": watt_total_low,
+#               "watt_total_low": watt_total_high
+#           }
+#       }
+#     ]
+#     current_wattages = []
+#     client.write_points(json_body)
 
 
+class DmsrFourTelegram:
+    current_usage_electricity: float
+    total_usage_electricity_low: float
+    total_usage_electricity_high: float
+
+class DsmrInfo:
+
+    def __init__(self, variable: str, search_for: str,  unit: str):
+        self.variable_name = variable
+        self.search_for = search_for
+        self.unit = unit
+    variable_name: str
+    search_for: str
+    unit: str
+
+class DsmrFour():
+    _serial : serial.Serial
+    lines: List[str] = []
+
+    _registrations: List[DsmrInfo] = []
+    def __init__(self, ser: serial.Serial):
+        ser.baudrate = 115200
+        ser.bytesize =serial.EIGHTBITS
+        ser.parity = serial.PARITY_NONE
+        ser.stopbits = serial.STOPBITS_ONE
+        ser.xonxoff = 0
+        ser.rtscts = 0
+        ser.timeout = 12
+        ser.port = "/dev/ttyUSB0"
+        self._serial = ser
+    
+    def register(self,data_unit: DsmrInfo)  -> None:
+        self._registrations.append(data_unit)
+
+    def interpret(self) -> None:
+        for finder in self._registrations:
+            for text in self.lines:
+                if finder.search_for in text:
+                    groups = re.findall(r'\((.*?)\)',text)
+
+                    text = [group.replace(f"*{finder.unit}", "") for group in groups if str(group).endswith(f"*{finder.unit}")][0]
```

Comment:
Why not include the `endswith` matching part in the regex if you only want the results from lines matching that?

Reply:
Now that it's in the regex, you don't need the if condition here anymore, right?